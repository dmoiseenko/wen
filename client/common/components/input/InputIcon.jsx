import React from 'react';
import PropTypes from 'prop-types';


export default function Icon({ iconClassName, error }) {
  return (
    <span className="icon">
      {
        error ?
          <i className="fa fa-warning" />
          :
          <i className={iconClassName} />
      }
    </span>
  );
}

Icon.propTypes = {
  iconClassName: PropTypes.string.isRequired,
  error: PropTypes.bool.isRequired
};

Icon.defaultProps = {
  error: false
};
