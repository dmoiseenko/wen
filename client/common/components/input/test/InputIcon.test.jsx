import React from 'react';
import renderer from 'react-test-renderer';

import InputIcon from '../InputIcon';


it('should render without error', () => {
  const props = {
    iconClassName: 'iconClassName',
    error: false
  };

  const tree = renderer.create(
    <InputIcon {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render with error', () => {
  const props = {
    iconClassName: 'iconClassName',
    error: true
  };

  const tree = renderer.create(
    <InputIcon {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
