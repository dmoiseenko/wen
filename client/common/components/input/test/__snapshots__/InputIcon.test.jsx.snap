// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with error 1`] = `
<span
  className="icon"
>
  <i
    className="fa fa-warning"
  />
</span>
`;

exports[`should render without error 1`] = `
<span
  className="icon"
>
  <i
    className="iconClassName"
  />
</span>
`;
